pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
+			name:                  "userset_group_granularity_with_direct_user_relationships",
+			TemporarilySkipReason: "because `group:eng` is being returned instead of `group:fga#member`",
 			req: &openfgav1.ListUsersRequest{
-				StoreId:              storeID,
-				AuthorizationModelId: modelID,
-				Object:               &openfgav1.Object{Type: "document", Id: "1"},
-				Relation:             "viewer",
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type:     "group",
+						Relation: "member",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user]
+			type document
+				relations
+					define viewer: [ user, group#member ]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("group:eng", "member", "user:will"),
+				tuple.NewTupleKey("group:eng", "member", "user:maria"),
+				tuple.NewTupleKey("group:marketing", "member", "user:jon"),
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+
+				tuple.NewTupleKey("document:1", "viewer", "user:poovam"),
+			},
+			expectedUsers: []string{"group:eng#member"},
+		},
+		{
+			name: "userset_multiple_usersets",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
 				UserFilters: []*openfgav1.ListUsersFilter{
 					{
 						Type: "user",
 					},
 				},
 			},
-			typeDefinitions: parser.MustParse(`
+			model: `model
+            schema 1.1
 			type user
 			type group
 			  relations
-			    define member: [user, group#member] as self
+			    define member: [user, group#member]
 			type document
 			  relations
-			    define viewer: [group#member] as self
-			`),
+			    define viewer: [group#member]`,
 			tuples: []*openfgav1.TupleKey{
-				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
 				tuple.NewTupleKey("group:eng", "member", "user:hawker"),
+				tuple.NewTupleKey("group:fga", "member", "user:jon"),
 				tuple.NewTupleKey("group:eng", "member", "group:fga#member"),
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+			},
+			expectedUsers: []string{"user:jon", "user:hawker"},
+		},
+		{
+			name:                  "userset_multiple_usersets_group_granularity",
+			TemporarilySkipReason: "because `group:eng` and `group:fga` is being returned instead of `group:eng#member` and `group:fga#member`",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type:     "group",
+						Relation: "member",
+					},
+				},
+			},
+			model: `model
+            schema 1.1
+			type user
+			type group
+			  relations
+			    define member: [user, group#member]
+			type document
+			  relations
+			    define viewer: [group#member]`,
+			tuples: []*openfgav1.TupleKey{
```

Comment:
Similar comment to that above, can we also add one more at the `document:1#viewer` level just for good measure?

* document:1#viewer@group:eng#member
* document:1#viewer@group:other#member
* ...

expected: [`group:eng#member`, `group:fga#member`, `group:other#member`]